### PR TITLE
UCT/TCP: is_connected implementation for TCP

### PR DESCRIPTION
**Pull request by shasson5**
## What
is_connected implementation for TCP

## Why ?
Support is_connected API

### CODE REVIEW

File: src/uct/tcp/tcp_ep.c [hunk 1]
```diff
@@ -2163,3 +2163,28 @@ uct_tcp_ep_check(uct_ep_h tl_ep, unsigned flags, uct_completion_t *comp)
...
+    UCT_EP_PARAMS_CHECK_IS_CONNECTED_DEV_IFACE_ADDRS(params);
+
+    status = uct_tcp_ep_set_dest_addr(params->device_addr, params->iface_addr,
+                                      (struct sockaddr*)&dest_addr);
+    if (status != UCS_OK) {
+        return 0;
+    }
+
+    is_connected = !ucs_sockaddr_cmp((struct sockaddr*)&ep->peer_addr,
+                                     (struct sockaddr*)&dest_addr, &status);
```
Comment (yosefe) [inline src/uct/tcp/tcp_ep.c:2184]: add const to both args

File: src/uct/tcp/tcp_ep.c [hunk 2]
```diff
@@ -2163,3 +2163,28 @@ uct_tcp_ep_check(uct_ep_h tl_ep, unsigned flags, uct_completion_t *comp)
...
+    if (status != UCS_OK) {
```
Comment (yosefe) [inline src/uct/tcp/tcp_ep.c:2185]: how could status be != UCS_OK here?

File: src/uct/tcp/tcp.h [hunk 3]
```diff
@@ -673,4 +673,7 @@ ucs_status_t uct_tcp_query_devices(uct_md_h md,
                                    uct_tl_device_resource_t **devices_p,
                                    unsigned *num_devices_p);
 
+int uct_tcp_ep_is_connected(uct_ep_h tl_ep,
+                            const uct_ep_is_connected_params_t *param);
```
Comment (yosefe) [inline src/uct/tcp/tcp.h:677]: param -> params

File: src/uct/base/uct_iface.h [hunk 4]
```diff
@@ -115,6 +115,42 @@ enum {
         return UCS_ERR_INVALID_PARAM; \
     }
 
+/**
+ * Returns 0 if @a _params field mask does not have the required flags set.
+ */
+#define UCT_EP_PARAMS_CHECK_ADDRS(_params, _flags, _err_message, ...) \
```
Comment (yosefe) [inline src/uct/base/uct_iface.h:121]: the name of this macro does not match  the documentation

File: src/uct/base/uct_iface.h [hunk 5]
```diff
@@ -115,6 +115,42 @@ enum {
...
+        ucs_error(_err_message, ##__VA_ARGS__); \
+        return 0; \
+    }
+
+/**
+ * Returns 0 if @a _params field mask does not have
+ * @ref UCT_EP_IS_CONNECTED_FIELD_DEVICE_ADDR and @ref UCT_EP_IS_CONNECTED_FIELD_IFACE_ADDR
+ * flags set.
+ */
+#define UCT_EP_PARAMS_CHECK_IS_CONNECTED_DEV_IFACE_ADDRS(_params) \
```
Comment (yosefe) [inline src/uct/base/uct_iface.h:132]: UCT_EP_IS_CONNECTED_CHECK_DEV_IFACE_ADDRS

File: src/uct/base/uct_iface.h [hunk 6]
```diff
@@ -115,6 +115,42 @@ enum {
...
+            "missing params (field_mask: %lu), both device_addr and " \
+            "iface_addr must be provided.", \
+            params->field_mask)
+
+/**
+ * Returns 0 if @a _params field mask does not have
+ * @ref UCT_EP_IS_CONNECTED_FIELD_DEVICE_ADDR and @ref UCT_EP_IS_CONNECTED_FIELD_IFACE_ADDR
+ * flags set.
+ */
+#define UCT_EP_PARAMS_CHECK_IS_CONNECTED_DEV_EP_ADDRS(_params) \
```
Comment (yosefe) [inline src/uct/base/uct_iface.h:146]: macro is unused in this PR

File: src/uct/tcp/tcp_ep.c (hunk 2 above)
Comment (shasson5) [reply src/uct/tcp/tcp_ep.c:2185]: if ucs_sockaddr_cmp returned UCS_ERR_INVALID_PARAM

File: src/uct/base/uct_iface.h (hunk 4 above)
Comment (shasson5) [reply src/uct/base/uct_iface.h:121]: you mean not specific enough?

File: src/uct/base/uct_iface.h [hunk 7]
```diff
@@ -118,7 +118,7 @@ enum {
 /**
  * Returns 0 if @a _params field mask does not have the required flags set.
  */
-#define UCT_EP_PARAMS_CHECK_ADDRS(_params, _flags, _err_message, ...) \
+#define UCT_EP_PARAMS_CHECK_FLAGS(_params, _flags, _err_message, ...) \
```
Comment (yosefe) [inline src/uct/base/uct_iface.h:121]: UCT_PARAMS_CHECK_FIELD_MASK(_params, _required_fields, _err_message, ...)

**Review by yosefe — APPROVED** [review]
pls squash

**Pull request merged**